drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -531,31 +770,29 @@ def _get_grade(self):
         correct_count, required_count = self._get_item_stats()
         return correct_count / float(required_count) * self.weight
 
-    def _is_finished(self):
+    def _answer_correctness(self):
         """
-        All items are at their correct place and a value has been
-        submitted for each item that expects a value.
+        Checks answer correctness:
+
+        Returns:
+            string: Correct/Incorrect/Partial
+                * Correct: All items are at their correct place.
+                * Partial: Some items are at their correct place.
+                * Incorrect: None items are at their correct place.
         """
         correct_count, required_count = self._get_item_stats()
-        return correct_count == required_count
-
-    @XBlock.json_handler
-    def publish_event(self, data, suffix=''):
-        try:
-            event_type = data.pop('event_type')
-        except KeyError:
-            return {'result': 'error', 'message': 'Missing event_type in JSON data'}
-
-        self.runtime.publish(self, event_type, data)
-        return {'result': 'success'}
+        if correct_count == required_count:
+            return self.SOLUTION_CORRECT
+        elif correct_count == 0:
+            return self.SOLUTION_INCORRECT
+        else:
+            return self.SOLUTION_PARTIAL
 
-    def _get_unique_id(self):
-        usage_id = self.scope_ids.usage_id
-        try:
-            return usage_id.name
-        except AttributeError:
-            # workaround for xblock workbench
-            return usage_id
+    def _is_finished(self):
```

Comment:
I actually found it confusing above (line 663) that a problem is not considered "finished" in assessment mode unless all attempts are exhausted, even if the answer is correct. If the desire is truly to have these different concepts of "finished", it would be helpful to rename this method-- _is_finished_standard_mode or something similar.